internal/apiserver/createInvoice.go

```diff
@@ -88,69 +35,80 @@ func validateInvoice(ctx goContext.Context, db *db.Connector, inv *entity.Invoic
 	}
 }
 
-func (a *App) createInvoice(res http.ResponseWriter, req *http.Request) error {
-	requestBody := CreateInvoiceRequestBody{}
-	err := requestBody.parse(req)
-	if err != nil {
-		return err
-	}
-	userId := context.GetUserId(req.Context())
+func (a *App) createInvoice(testInvoice bool) func(res http.ResponseWriter, req *http.Request) error {
+	return func(res http.ResponseWriter, req *http.Request) error {
+		invoice, err := parseInvoice(req)
+		if err != nil {
+			return InvoiceError("invoice.parsingError").WithDetail(err)
+		}
 
-	// Validate invoice format
-	if err = a.xsdValidator.Validate(requestBody.invoice, requestBody.format, requestBody.documentType); err != nil {
-		return InvoiceError("xsd.validation.failed").WithDetail(err)
-	}
-	// In future possibly adjust validation according to partiesType
-	if err = a.invoiceValidator.Validate(req.Context(), requestBody.invoice, requestBody.format, requestBody.language); err != nil {
-		if _, ok := err.(*invoiceValidator.ValidationError); ok {
-			return handlerutil.NewBadRequestError("invoice.validation.failed").WithDetail(err)
-		} else if _, ok := err.(*invoiceValidator.RequestError); ok {
-			return handlerutil.NewFailedDependencyError("invoice.validation.request.failed")
-		} else {
-			return err
+		format, documentType, err := a.xsdValidator.GetFormatAndType(invoice)
+		if err != nil {
+			return InvoiceError("invoice." + err.Error())
```

Comment:
`GetFormatAndType` can return weird errors (from xml parsing), let's return error as detail and use single message like `invoice.format.invalid`